packages/devtools_app/lib/src/memory/memory_android_chart.dart

```diff
@@ -0,0 +1,324 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_shared/devtools_shared.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../auto_dispose_mixin.dart';
+import '../charts/chart.dart';
+import '../charts/chart_controller.dart';
+import '../charts/chart_trace.dart' as trace;
+import '../theme.dart';
+
+import 'memory_controller.dart';
+import 'memory_timeline.dart';
+
+class MemoryAndroidChart extends StatefulWidget {
+  const MemoryAndroidChart(this.chartController);
+
+  final ChartController chartController;
+
+  @override
+  MemoryAndroidChartState createState() => MemoryAndroidChartState();
+}
+
+/// Name of each trace being charted, index order is the trace index
+/// too (order of trace creation top-down order).
+enum TraceName {
+  stack,
+  graphics,
+  nativeHeap,
+  javaHeap,
+  code,
+  other,
+  system,
+  total,
+}
+
+class MemoryAndroidChartState extends State<MemoryAndroidChart>
+    with AutoDisposeMixin {
+  /// Controller attached to the chart.
+  ChartController get _chartController => widget.chartController;
+
+  /// Controller for managing memory collection.
+  MemoryController _memoryController;
+
+  MemoryTimeline get _memoryTimeline => _memoryController.memoryTimeline;
+
+  ColorScheme colorScheme;
+
+  @override
+  void initState() {
+    setupTraces();
```

Comment:
i thought the parent initState should be called first. @pq is there a lint for this?

Reply:
I don't believe so.  The closest thing we have is the `@mustCallSuper` annotation but it doesn't require you call `super.*` first...